### PR TITLE
feat(orderbook): wire maker reputation from the Kind 38383 rating tag

### PR DESCRIPTION
**Pull request by grunch**
## Summary

Fixes the order book showing **0.0 rating / 0 trades / 0 days** for every live order. This gap predates the UI redesign (#243): when the real Kind 38383 bridge replaced the phase-5 demo data (`8bc109f`), `OrderInfo` never gained reputation fields and the parser silently dropped the daemon's `rating` tag — so both the order-book card and the take-order "Creator reputation" block have rendered zeros ever since.

### What changed

- **`rust/src/nostr/order_events.rs`**: `parse_order_event` now parses the `rating` tag the daemon publishes on each 38383 order (mostro-core `Rating` snapshot). Accepted shapes, mirroring the v1 client:
  - plain JSON object `{"total_reviews":47,"total_rating":4.9,...,"days":312}`
  - legacy array wrapping `["rating", {…}]`
  - `"none"` (full-privacy maker) → zeros
  - missing tag or malformed JSON → zeros, **without dropping the order**
- **`rust/src/api/types.rs`**: `OrderInfo` gains `rating: f64`, `total_reviews: u32`, `days_active: u32` (documented; zeros mean "no reputation to show").
- **`lib/features/home/providers/home_order_providers.dart`**: `OrderItem.fromInfo` maps the three new fields, so the redesigned order-book card and `take_order_screen`'s reputation block now display real maker reputation with no further UI change.
- flutter_rust_bridge bindings regenerated (pinned codegen 2.11.1).

### Tests

- 5 new Rust parser cases: object form, array-wrapped form, `"none"`, malformed JSON (order must survive), missing tag.
- New Dart test `order_item_reputation_test.dart` covering the `fromInfo` mapping (real values and full-privacy zeros).
- `cargo test` 143/143, `cargo clippy` clean, `flutter analyze` clean, `flutter test` 158/158.

## Manual testing guide

1. `flutter run` against a Mostro node with active makers (or use mostro-cli to publish an order from an identity with completed trades).
2. Open the Order Book: cards from makers with history must show non-zero `★ rating · N trades · N days` matching the `rating` tag content of their 38383 events (inspect with any Nostr relay browser).
3. Orders from full-privacy makers (`rating: none`) and freshly created identities must still render `0 · 0 trades · 0 days` and must not disappear from the book.
4. Tap an order → the "Creator reputation" card on the take-order screen shows the same values.
5. Regression: creating your own order still works (own orders carry zeros until the daemon's confirmation event arrives with the real snapshot).

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01VQUKX88xRBCPsHQTXBEiYa

## Summary by CodeRabbit

- **New Features**
  - Order listings now show maker reputation details: rating, total review count, and activity duration.
  - Reputation data is supported end-to-end through newly created/updated orders and incoming order events.

- **Bug Fixes**
  - Fixed order-to-UI mapping so reputation fields populate correctly instead of defaulting.

- **Tests**
  - Added widget and parsing coverage for valid reputation values, private/none handling, missing tags, malformed input, and out-of-range/boundary cases.

### CODE REVIEW

**Comment by grunch** [comment]
@coderabbitai review

File: rust/src/api/types.rs [hunk 1]
```diff
@@ -210,6 +210,14 @@ pub struct OrderInfo {
...
     pub expires_at: Option<i64>,
     pub is_mine: bool,
+    /// Maker reputation from the Kind 38383 `rating` tag (`total_rating`
+    /// aggregate, 0–5). `0.0` when the maker has no reputation yet or
+    /// publishes in full-privacy mode (`rating` = `"none"`).
+    pub rating: f64,
+    /// Number of reviews behind [`Self::rating`] (`total_reviews`).
+    pub total_reviews: u32,
+    /// Days the maker has been active on this Mostro node (`days`).
+    pub days_active: u32,
```
[the remark this answers is not in the document]
Comment (grunch) [reply rust/src/api/types.rs:227]: Fixed in e5d1a37 — the three fields now carry `#[serde(default)]`, so orders-table rows and `OrderInfo` nested in trades JSON persisted before this PR keep deserializing with zeros after an upgrade. Added a regression test that loads a legacy `OrderInfo` JSON without the fields.

[the remark this answers is not in the document]
Comment (grunch) [reply rust/src/api/types.rs:227]: Fixed in e5d1a37 — `specs/004-mostro-p2p-client/data-model.md` Order contract now lists `rating`, `total_reviews`, and `days_active` with their sources, plus the 0–5 validation rule (out-of-range values degrade to 0 without rejecting the order).

File: rust/src/nostr/order_events.rs [hunk 2]
```diff
@@ -90,9 +92,47 @@ pub fn parse_order_event(event: &Event, my_pubkey: Option<&PublicKey>) -> Option
...
+        {
+            arr[1].as_object().expect("checked is_object above")
+        }
+        _ => return EMPTY,
+    };
+    let num = |key: &str| obj.get(key).and_then(|v| v.as_f64()).unwrap_or(0.0);
+    (
+        num("total_rating"),
+        num("total_reviews") as u32,
+        num("days") as u32,
```
[the remark this answers is not in the document]
Comment (grunch) [reply rust/src/nostr/order_events.rs]: Fixed in e5d1a37 — applied the suggested validation: `total_rating` must be within 0–5 (`filter`), counts go through `as_u64` + `u32::try_from` so fractional, negative, or oversized values fall back to that field's zero default independently. Added boundary tests: out-of-range/fractional/negative mix, exact 0 and 5 bounds, and a `total_reviews` value above u32::MAX.

File: specs/004-mostro-p2p-client/data-model.md [hunk 3]
```diff
@@ -51,12 +51,16 @@ A buy or sell offer on the Mostro network.
...
+| rating | f64 | Maker reputation from the Kind 38383 `rating` tag (`total_rating`, 0–5; 0.0 = none/full privacy) |
+| total_reviews | u32 | Number of reviews behind `rating` (`total_reviews`) |
+| days_active | u32 | Days the maker has been active on the node (`days`) |
 
 **Validation rules**:
 - `fiat_code` MUST be a valid ISO 4217 code.
 - Either `fiat_amount` OR both `fiat_amount_min` and `fiat_amount_max` MUST be provided, but NOT both. If `fiat_amount` is present, `fiat_amount_min` and `fiat_amount_max` MUST be absent; if `fiat_amount_min`/`fiat_amount_max` are present, `fiat_amount` MUST be absent.
 - If range: `fiat_amount_min` MUST be > 0 and < `fiat_amount_max`.
 - `premium` is a signed float (negative = discount).
+- `rating` MUST be within 0–5; out-of-range or malformed `rating`-tag values degrade to 0 without rejecting the order.
```
[the remark this answers is not in the document]
Comment (grunch) [reply specs/004-mostro-p2p-client/data-model.md]: Fixed in 7d5ec0d — the `rating` field description now states 0.0 covers full privacy (`none`), a missing tag, and malformed/invalid data; the validation rule spells out that each reputation field (`rating`, `total_reviews`, `days_active`) degrades to 0 independently and the order is never rejected because of its `rating` tag — matching `parse_rating_tag`'s behavior.
